### ISSUE TITLE
Issue: Mislabeled fields due to parsing error of content type json

### ISSUE THREAD

**Issue by anoduck** [estruyf/vscode-front-matter#868]
**Describe the bug**

For some reason, FrontMatter is misinterpreting the contenttype configuration file for the field `"thumbnail":`. The thumbnail field is "heading field" or "primary field", which has several "subfields" or "secondary fields" on the next level below it. (The jargon is a little vague, admittedly) All of these subfields are strings, but FrontMatter is displaying all of them as datetime fields except the "authorURL" subfield, oddly that one is correct. 

Below, I have attached the portion of the json file I am referring to. I have also uploaded a screenshot of what I am referring to, and the complete json file for that content type. Maybe I have misconfigured my json file.

```json
{
            "title": "thumbnail",
            "name": "thumbnail",
            "type": "fields",
            "fields": [
                {
                    "title": "url",
                    "name": "url",
                    "type": "string"
                },
                {
                    "title": "author",
                    "name": "author",
                    "type": "string"
                },
                {
                    "title": "authorURL",
                    "name": "authorURL",
                    "type": "string"
                },
                {
                    "title": "origin",
                    "name": "origin",
                    "type": "string"
                },
                {
                    "title": "originURL",
                    "name": "originURL",
                    "type": "string"
                }
            ]
        }
```

**To Reproduce**

1. Setup FrontMatter for a hugo site.
2. Use the json file provided below as one of the content types
3. Create content using that content type.

**Expected behavior**

All the subfields should be string fields, preferably single lined.

**Screenshots**

![FrontMatter-bug](https://github.com/user-attachments/assets/dd126c32-6510-4a61-80ef-0296b352fd4e)

**Device:**

- OS: Kali Linux
- Front Matter CMS Version: Front Matter (v10.5.112333718-beta BETA)
- Browser N/A, but FireFox

**Additional context**

This is the configuration file for that content type:

```json
{
    "$schema": "https://frontmatter.codes/config/taxonomy.contenttypes.schema.json",
    "name": "legal",
    "pageBundle": false,
    "fields": [
        {
            "title": "title",
            "name": "title",
            "type": "string",
            "single": true
        },
        {
            "title": "description",
            "name": "description",
            "type": "string"
        },
        {
            "title": "date",
            "name": "date",
            "type": "datetime",
            "isPublishDate": true
        },
        {
            "title": "tags",
            "name": "tags",
            "type": "tags"
        },
        {
            "title": "Categories",
            "name": "categories",
            "type": "choice",
            "choices": [
                "Persons",
                "Evidence",
                "Legal"
            ]
        },
        {
            "title": "Type",
            "name": "type",
            "type": "choice",
            "choices": [
                "default",
                "docs",
                "minimal"
            ],
            "default": "docs"
        },
        {
            "title": "draft",
            "name": "draft",
            "type": "draft"
        },
        {
            "title": "thumbnail",
            "name": "thumbnail",
            "type": "fields",
            "fields": [
                {
                    "title": "url",
                    "name": "url",
                    "type": "string"
                },
                {
                    "title": "author",
                    "name": "author",
                    "type": "string"
                },
                {
                    "title": "authorURL",
                    "name": "authorURL",
                    "type": "string"
                },
                {
                    "title": "origin",
                    "name": "origin",
                    "type": "string"
                },
                {
                    "title": "originURL",
                    "name": "originURL",
                    "type": "string"
                }
            ]
        }
    ]
}
```

**Comment by anoduck** [comment]
Found it! Along with having issues regarding uniformity of labels across types, in another content type file, the subfields were assigned to datetime types. So, this other file must have been created before the one mentioned above.

**Comment by estruyf** [comment]
Hi @anoduck, I've checked your configuration, but I don't see anything wrong with it, and it outputs the correct fields in my test setup:

<img width="344" alt="image" src="https://github.com/user-attachments/assets/8f351e55-f9e5-4d27-add6-ab2e49947558">

The only thing I see, is that your `origin-url` and `author-url` properties are not correctly mapped in the content type. The name `originURL` and `authorURL` need to be updated. Although, this should not cause it from rendering as `datetime` fields.

**Comment by estruyf** [comment]
Oh, thanks @anoduck for the update.

**Comment by anoduck** [comment]
@estruyf I know it has to be frustrating for maintainers, which is why maintainers are the superheroes of the Open Source community. 

I had returned to VSCode and by chance discovered a discrepancy in the labeling of a pair of fields. It was during the correction of these discrepancies, I discovered a contenttype which actually had the above-mentioned values defined as a datetime type. Although, this setting was not in the same contenttype referenced in the example above, I believe it established a precedence in frontMatter's database that lead to the creation of this issue.

Blessings.
